src/loaders/features/featureDependencies.js

```diff
@@ -0,0 +1,26 @@
+import { FEATURE_NAMES } from "./features"
+
+export function getFeatureDependencyNames(feature) {
+    switch (feature) {
+        case FEATURE_NAMES.ajax:
+            return [FEATURE_NAMES.jserrors]
+        case FEATURE_NAMES.sessionTrace:
+            return [FEATURE_NAMES.ajax, FEATURE_NAMES.pageViewEvent]
+        case FEATURE_NAMES.pageViewTiming:
+            return [FEATURE_NAMES.pageViewEvent] // this could change if we disconnect window load timings
+        default:
+            return []
+    }
+}
+
+export function getFrozenAttributes(feature) {
+    switch (feature) {
+        // right now, jserrors is the only feature that can have "on" or "off" page-level auto-instrumentation...
+        // page_action is always "off" (no instr)
+        // as new API/manual implementation methods are added, this list can likely be pruned
+        case FEATURE_NAMES.jserrors:
+            return []
+        default:
+            return ['auto']
```

Comment:
In relation to the issue I pointed out in _instrument-base.js_ about `auto` property, this is what concerns me: that customers can redefine it as off in the `init` block and then they no longer follow the drain-gating logic.

Reply:
thats a different auto.  This is a remnant that can be removed

Reply:
https://issues.newrelic.com/browse/NEWRELIC-6480